hub/api/objectview.py

```diff
@@ -0,0 +1,373 @@
+        #     self.num_process(schema_obj.dtype)
+
+    def process_path(self, subpath, inner_schema_obj, nums, offsets, squeeze_dims):
+        paths = subpath.split("/")[1:]
+        try:
+            if inner_schema_obj:
+                if isinstance(inner_schema_obj, Sequence):
+                    schema_obj = inner_schema_obj.dtype.dict_[paths[0]]
+                elif isinstance(inner_schema_obj, SchemaDict):
+                    schema_obj = inner_schema_obj.dict_[paths[0]]
+                else:
+                    raise KeyError()
+            else:
+                schema_obj = self.schema[paths[0]]
+        except (KeyError, AttributeError):
+            raise KeyError(f"{paths[0]} is an invalid key")
+        self.num_process(schema_obj, nums, offsets, squeeze_dims)
+        for path in paths[1:]:
+            try:
+                if isinstance(schema_obj, Sequence):
+                    schema_obj = schema_obj.dtype.dict_[path]
+                elif isinstance(schema_obj, SchemaDict):
+                    schema_obj = schema_obj.dict_[path]
+                else:
+                    raise KeyError()
+                self.num_process(schema_obj, nums, offsets, squeeze_dims)
+            except (KeyError, AttributeError):
+                raise KeyError(f"{path} is an invalid key")
+        return schema_obj, nums, offsets, squeeze_dims
+
+    def __getitem__(self, slice_):
+        if not isinstance(slice_, abc.Iterable) or isinstance(slice_, str):
+            slice_ = [slice_]
+        slice_ = list(slice_)
+        subpath, slice_list = slice_split(slice_)
+
+        dataset = self.dataset
+        nums, offsets, squeeze_dims, inner_schema_obj = (
+            self.nums.copy(),
+            self.offsets.copy(),
+            self.squeeze_dims.copy(),
+            self.inner_schema_obj,
+        )
+
+        if subpath:
+            inner_schema_obj, nums, offsets, squeeze_dims = self.process_path(
+                subpath, inner_schema_obj, nums, offsets, squeeze_dims
+            )
+        subpath = self.subpath + subpath
+        if len(slice_list) >= 1:
+            # Slice first dim
+            if isinstance(self.dataset, DatasetView) and not self.dataset.squeeze_dim:
+                dataset = self.dataset[slice_list[0]]
+                slice_list = slice_list[1:]
+            elif not isinstance(self.dataset, DatasetView):
+                num, ofs = slice_extract_info(slice_list[0], self.dataset.shape[0])
+                dataset = DatasetView(
+                    self.dataset, num, ofs, isinstance(slice_list[0], int)
+                )
+                slice_list = slice_list[1:]
+
+            # Expand slice list for rest of dims
+            if len(slice_list) >= 1:
+                exp_slice_list = []
+                for squeeze in squeeze_dims:
+                    if squeeze:
+                        exp_slice_list += [None]
+                    else:
+                        try:
```

Comment:
Add explicit checks for size of list rather than catching exceptions.

Reply:
Done